### ISSUE TITLE
storage: ntfs not support

### ISSUE THREAD

**Issue by yizhanglinux** [linux-system-roles/storage#151]
# environment
```
# uname -r
5.6.6-300.fc32.x86_64
# cat /etc/redhat-release
Fedora release 32 (Thirty Two)
```
# playbook
```
# cat tests/ntfs_disk.yml 
- hosts: all
  vars:
    mount_location: '/opt/test1'
    volume_group_size: '10g'
    volume_size_before: '5g'
    volume_size_after: '9g'

  tasks:
    - include_role:
        name: storage

    - include_tasks: get_unused_disk.yml
      vars:
        min_size: "{{ volume_group_size }}"
        max_return: 1

    - name: Create one ntfs fs disk 
      include_role:
        name: storage
      vars:
          storage_volumes:
            - name: foo
              disks:
                - sdf
              type: disk 
              fs_type: 'ntfs'
              mount_point: "{{ mount_location }}"
```
# execution log
```
---snip---
TASK [storage : manage the pools and volumes to match the specified state] ****************************************************************************************************************************************************************************************************************************************************************************************************************
task path: /root/test/storage/tasks/main-blivet.yml:104
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: root
<localhost> EXEC /bin/sh -c 'echo ~root && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp `"&& mkdir /root/.ansible/tmp/ansible-tmp-1597658552.6994646-28894-122065178217497 && echo ansible-tmp-1597658552.6994646-28894-122065178217497="` echo /root/.ansible/tmp/ansible-tmp-1597658552.6994646-28894-122065178217497 `" ) && sleep 0'
Using module file /root/test/storage/library/blivet.py
<localhost> PUT /root/.ansible/tmp/ansible-local-28512dx15fz8s/tmp26hbxlib TO /root/.ansible/tmp/ansible-tmp-1597658552.6994646-28894-122065178217497/AnsiballZ_blivet.py
<localhost> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1597658552.6994646-28894-122065178217497/ /root/.ansible/tmp/ansible-tmp-1597658552.6994646-28894-122065178217497/AnsiballZ_blivet.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python3 /root/.ansible/tmp/ansible-tmp-1597658552.6994646-28894-122065178217497/AnsiballZ_blivet.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1597658552.6994646-28894-122065178217497/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
  File "/tmp/ansible_blivet_payload_y36wzto6/ansible_blivet_payload.zip/ansible/modules/blivet.py", line 1181, in run_module
  File "/tmp/ansible_blivet_payload_y36wzto6/ansible_blivet_payload.zip/ansible/modules/blivet.py", line 898, in manage_volume
  File "/tmp/ansible_blivet_payload_y36wzto6/ansible_blivet_payload.zip/ansible/modules/blivet.py", line 415, in manage
  File "/tmp/ansible_blivet_payload_y36wzto6/ansible_blivet_payload.zip/ansible/modules/blivet.py", line 447, in _create
  File "/tmp/ansible_blivet_payload_y36wzto6/ansible_blivet_payload.zip/ansible/modules/blivet.py", line 393, in _reformat
  File "/tmp/ansible_blivet_payload_y36wzto6/ansible_blivet_payload.zip/ansible/modules/blivet.py", line 338, in _get_format
fatal: [localhost]: FAILED! => {
    "actions": [],
    "changed": false,
    "crypts": [],
    "invocation": {
        "module_args": {
            "disklabel_type": null,
            "packages_only": false,
            "pools": [],
            "safe_mode": true,
            "use_partitions": null,
            "volumes": [
                {
                    "disks": [
                        "sdf"
                    ],
                    "encryption": false,
                    "encryption_cipher": null,
                    "encryption_key": null,
                    "encryption_key_size": null,
                    "encryption_luks_version": null,
                    "encryption_password": null,
                    "fs_create_options": "",
                    "fs_label": "",
                    "fs_overwrite_existing": true,
                    "fs_type": "ntfs",
                    "mount_check": 0,
                    "mount_device_identifier": "uuid",
                    "mount_options": "defaults",
                    "mount_passno": 0,
                    "mount_point": "/opt/test1",
                    "name": "foo",
                    "raid_chunk_size": null,
                    "raid_device_count": null,
                    "raid_level": null,
                    "raid_metadata_version": null,
                    "raid_spare_count": null,
                    "size": 0,
                    "state": "present",
                    "type": "disk"
                }
            ]
        }
    },
    "leaves": [],
    "mounts": [],
    "msg": "required tools for file system 'ntfs' are missing",
    "packages": [],
    "pools": [],
    "volumes": []
}

PLAY RECAP ********************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=31   changed=0    unreachable=0    failed=1    skipped=19   rescued=0    ignored=0   

```
# cat /tmp/blivet.log | tail -20
```
2020-08-17 06:02:35,028 INFO program/MainThread: Running... udevadm settle --timeout=300
2020-08-17 06:02:35,051 DEBUG program/MainThread: Return code: 0
2020-08-17 06:02:35,078 INFO blivet/MainThread: edd: MBR signature on sda is zero. new disk image?
2020-08-17 06:02:35,078 INFO blivet/MainThread: edd: MBR signature on sdl is zero. new disk image?
2020-08-17 06:02:35,078 INFO blivet/MainThread: edd: collected mbr signatures: {}
2020-08-17 06:02:35,083 DEBUG blivet/MainThread:                DeviceTree.get_device_by_path: path: /dev/mapper/fedora_storageqe--62-root ; incomplete: False ; hidden: False ;
2020-08-17 06:02:35,086 DEBUG blivet/MainThread:                DeviceTree.get_device_by_path returned existing 15 GiB lvmlv fedora_storageqe-62-root (43) with existing xfs filesystem
2020-08-17 06:02:35,086 DEBUG blivet/MainThread: resolved '/dev/mapper/fedora_storageqe--62-root' to 'fedora_storageqe-62-root' (lvmlv)
2020-08-17 06:02:35,087 DEBUG blivet/MainThread: resolved 'UUID=439fbba7-c553-44d9-ac42-3cad0a46290e' to 'sda2' (partition)
2020-08-17 06:02:35,087 DEBUG blivet/MainThread: resolved 'UUID=1328-582B' to 'sda1' (partition)
2020-08-17 06:02:35,089 DEBUG blivet/MainThread:                DeviceTree.get_device_by_path: path: /dev/mapper/fedora_storageqe--62-swap ; incomplete: False ; hidden: False ;
2020-08-17 06:02:35,091 DEBUG blivet/MainThread:                DeviceTree.get_device_by_path returned existing 7.88 GiB lvmlv fedora_storageqe-62-swap (56) with existing swap
2020-08-17 06:02:35,092 DEBUG blivet/MainThread: resolved '/dev/mapper/fedora_storageqe--62-swap' to 'fedora_storageqe-62-swap' (lvmlv)
2020-08-17 06:02:35,092 DEBUG blivet/MainThread: failed to resolve 'UUID=ab31a5b5-d7fb-4c63-8946-eaefd30e1f52'
2020-08-17 06:02:35,094 DEBUG blivet/MainThread:                  DeviceTree.get_device_by_name: name: sdf ; incomplete: False ; hidden: False ;
2020-08-17 06:02:35,097 DEBUG blivet/MainThread:                  DeviceTree.get_device_by_name returned existing 931.51 GiB disk sdf (165)
2020-08-17 06:02:35,097 DEBUG blivet/MainThread: resolved 'sdf' to 'sdf' (disk)
2020-08-17 06:02:35,100 DEBUG blivet/MainThread:                    NTFS.supported: supported: False ;
2020-08-17 06:02:35,100 DEBUG blivet/MainThread: get_format('ntfs') returning NTFS instance with object id 185
2020-08-17 06:02:35,102 DEBUG blivet/MainThread:                 NTFS.supported: supported: False ;

```

**Comment by yizhanglinux** [comment]
And mkfs.ntfs on the entire disk will be failed if no "-K" provided.
```
[root@storageqe-62 storage]# mkfs.ntfs /dev/sdc
/dev/sdc is entire device, not just one partition.
Refusing to make a filesystem here!
[root@storageqe-62 storage]# echo $?
1
[root@storageqe-62 storage]# mkfs.ntfs -Ff /dev/sdc 
/dev/sdc is entire device, not just one partition.
mkntfs forced anyway.
Cluster size has been automatically set to 4096 bytes.
Creating NTFS volume structures.
mkntfs completed successfully. Have a nice day.
[root@storageqe-62 storage]# mkfs.ntfs -h

Usage: mkntfs [options] device [number-of-sectors]

Basic options:
    -f, --fast                      Perform a quick format
    -Q, --quick                     Perform a quick format
    -L, --label STRING              Set the volume label
    -C, --enable-compression        Enable compression on the volume
    -I, --no-indexing               Disable indexing on the volume
    -n, --no-action                 Do not write to disk

Advanced options:
    -c, --cluster-size BYTES        Specify the cluster size for the volume
    -s, --sector-size BYTES         Specify the sector size for the device
    -p, --partition-start SECTOR    Specify the partition start sector
    -H, --heads NUM                 Specify the number of heads
    -S, --sectors-per-track NUM     Specify the number of sectors per track
    -z, --mft-zone-multiplier NUM   Set the MFT zone multiplier
    -T, --zero-time                 Fake the time to be 00:00 UTC, Jan 1, 1970
    -F, --force                     Force execution despite errors

Output options:
    -q, --quiet                     Quiet execution
    -v, --verbose                   Verbose execution
        --debug                     Very verbose execution

Help options:
    -V, --version                   Display version
    -l, --license                   Display licensing information
    -h, --help                      Display this help

Developers' email address: ntfs-3g-devel@lists.sf.net
News, support and information:  http://tuxera.com
```

**Comment by richm** [comment]
@vojtechtrefny do we need to do anything to the storage role for ntfs support?

**Comment by vojtechtrefny** [comment]
No, the blivet change is all we need to support NTFS, the playbook from the issue now works with blivet 3.5.0 and newer (available since RHEL 8.8 and 9.2).
